src/main/java/teammates/ui/webapi/output/InstructorData.java

```diff
@@ -85,6 +93,25 @@ public void setJoinState(JoinState joinState) {
         this.joinState = joinState;
     }
 
+    public String getKey() {
+        return key;
+    }
+
+    public void setKey(List<InstructorAttributes> instructorAttributes) {
+        this.key = StringHelper.encrypt(instructorAttributes.stream()
+                .filter((InstructorAttributes instructorAttribute)
```

Comment:
Same as https://github.com/TEAMMATES/teammates/pull/9990#discussion_r399765701

Can we improve this?

Reply:
Same comment but I copy pasted herre for ease of reference

Hey Xiao Pu,
I think there were two alternatives I considered when doing this.
First, is to put the key generation inside the constructor. However, this will place the burden of hiding the key on the developer. Since Teammates has many contributors, I opted not to do this as people who use StudentsData in the future may not be aware of this and thus send the key.

Yea, theres a runtime issue, but I think this should be fine as its Java, so it should run faster and because the performance bottleneck is at the frontend side.

What do you think?